lib/resty/core/phase.lua

```diff
@@ -0,0 +1,57 @@
+local ffi = require 'ffi'
+local C = ffi.C
+local base = require "resty.core.base"
+local tostring = tostring
+
+ffi.cdef[[
+int ngx_http_lua_ffi_get_phase(ngx_http_request_t *r, char **err)
+]]
+
+
+local _M = {
+    _VERSION = base.version
+}
+
+
+local errmsg = base.get_errmsg_ptr()
+local context_lookup = {
+    ["1"] = "set",
+    ["2"] = "rewrite",
+    ["4"] = "access",
+    ["8"] = "content",
+    ["16"] = "log",
+    ["32"] = "header_filter",
+    ["64"] = "body_filter",
+    ["128"] = "timer",
+    ["256"] = "init_worker",
+    ["512"] = "balancer",
+    ["1024"] = "ssl_cert",
+    ["2048"] = "ssl_session_store",
+    ["4096"] = "ssl_session_fetch",
```

Comment:
Why use strings as the keys here? You should use numbers directly. Lua tables should be clever enough to use a hash table instead of a sparse array for this table.